msos/MixedStack.cs

```diff
@@ -167,17 +175,102 @@ class ThreadInfo
         public uint OSThreadId { get; set; }
         public ClrThread ManagedThread { get; set; }
         public string Detail { get; set; }
-
         public bool IsManagedThread { get { return ManagedThread != null; } }
     }
 
+    class UnifiedThread
+    {
+        public UnifiedThread(ThreadInfo info)
+        {
+            IsManagedThread = info.IsManagedThread;
+            Index = info.Index;
+            EngineThreadId = info.EngineThreadId;
+            OSThreadId = info.OSThreadId;
+            Detail = info.Detail;
+            Info = info;
+        }
+
+        public UnifiedThread(uint owningThreadId)
+        {
+            this.OSThreadId = owningThreadId;
+        }
+        public ThreadInfo Info { get; private set; }
+        public List<UnifiedStackFrame> StackTrace { get; protected set; }
+        public List<UnifiedBlockingObject> BlockingObjects { get; protected set; }
+
+        public bool IsManagedThread { get; protected set; }
+        public uint Index { get; set; }
+        public uint EngineThreadId { get; set; }
+        public uint OSThreadId { get; set; }
+        public string Detail { get; set; }
+    }
+
+    class UnifiedManagedThread : UnifiedThread
+    {
+        public UnifiedManagedThread(ThreadInfo info, List<UnifiedStackFrame> managedStack, List<UnifiedStackFrame> unManagedStack, List<UnifiedBlockingObject> blockingObjects) : base(info)
+        {
+            StackTrace = new List<UnifiedStackFrame>();
+
+            if (managedStack != null)
+            {
+                StackTrace.AddRange(managedStack);
+            }
+
+            if (unManagedStack != null)
+            {
+                StackTrace.AddRange(unManagedStack);
+            }
```

Comment:
Does that produce the correct call stack? It's not always the case that all managed frames are strictly before all unmanaged frames.